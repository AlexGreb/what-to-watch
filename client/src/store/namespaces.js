export const namespaces = {
  APP: `app`,
  MOVIES: `movies`,
  USER: `user`
};
