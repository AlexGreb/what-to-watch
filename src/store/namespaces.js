export const namespaces = {
  APP: `app`,
  MOVIES: `movies`
};
