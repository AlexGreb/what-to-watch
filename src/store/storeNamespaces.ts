export enum Namespaces {
  APP = `app`,
  MOVIES = `movies`,
  USER = `user`
};
