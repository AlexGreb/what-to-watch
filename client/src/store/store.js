import {configureStore} from '@reduxjs/toolkit';
import appReducer from './app/appReducer';
import moviesReducer from './movies/moviesReducer.js';
import userReducer from './user/userReducer.js';
import {namespaces} from './namespaces';
import {createApi} from '../api.js';


const api = createApi((...args) => store.dispatch(...args));

export const store = configureStore({
  reducer: {
    [namespaces.APP]: appReducer,
    [namespaces.MOVIES]: moviesReducer,
    [namespaces.USER]: userReducer
  },
  middleware: (getDefaultMiddleware) => {
    return getDefaultMiddleware({
      thunk: {
        extraArgument: api
      },
    });
  },
  devTools: true
  // eslint-disable-next-line no-undef
  // devTools: process.env.NODE_ENV !== `production`,
});

// export type State = ReturnType<typeof store.getState>
