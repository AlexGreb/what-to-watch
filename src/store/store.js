import {configureStore} from '@reduxjs/toolkit';
import appReducer from './app/appReducer.js';
import moviesReducer from './movies/moviesReducer.js';
import userReducer from './user/userReducer.js';
import {namespaces} from './namespaces.js';
import {createApi} from '../api.js';

const api = createApi((...args) => store.dispatch(...args));

export const store = configureStore({
  reducer: {
    [namespaces.APP]: appReducer,
    [namespaces.MOVIES]: moviesReducer,
    [namespaces.USER]: userReducer
  },
  middleware: (getDefaultMiddleware) => {
    return getDefaultMiddleware({
      thunk: {
        extraArgument: api
      },
    });
  },
  devTools: true
  // eslint-disable-next-line no-undef
  // devTools: process.env.NODE_ENV !== `production`,
});
