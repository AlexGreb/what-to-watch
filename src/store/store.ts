import {configureStore} from '@reduxjs/toolkit';
import appReducer from './app/appReducer';
import moviesReducer from './movies/moviesReducer.js';
import userReducer from './user/userReducer.js';
import {Namespaces} from './storeNamespaces';
import {createApi} from '../api.js';


const api = createApi((...args) => store.dispatch(...args));

export const store = configureStore({
  reducer: {
    [Namespaces.APP]: appReducer,
    [Namespaces.MOVIES]: moviesReducer,
    [Namespaces.USER]: userReducer
  },
  middleware: (getDefaultMiddleware) => {
    return getDefaultMiddleware({
      thunk: {
        extraArgument: api
      },
    });
  },
  devTools: true
  // eslint-disable-next-line no-undef
  // devTools: process.env.NODE_ENV !== `production`,
});

export type State = ReturnType<typeof store.getState>