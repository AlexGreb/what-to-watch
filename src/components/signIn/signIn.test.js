/**
 * @jest-environment jsdom
 */
import React from 'react';
import {render} from '@testing-library/react';
import SignIn from './signIn.jsx';
import {Provider} from 'react-redux';
import {store} from '../../store/store.js';

describe(`SignIn`, () => {
  it(`Should render correctly`, () => {
    const {asFragment} = render(<Provider store={store}><SignIn/></Provider>);
    expect(asFragment()).toMatchSnapshot();
  });
});
