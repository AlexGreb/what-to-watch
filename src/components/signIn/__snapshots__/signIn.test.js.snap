// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn Should render correctly 1`] = `
<DocumentFragment>
  <div
    class="user-page"
  >
    <header
      class="page-header user-page__head"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="main.html"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <h1
        class="page-title user-page__title"
      >
        Sign in
      </h1>
    </header>
    <div
      class="sign-in user-page__content"
    >
      <form
        action="/login"
        class="sign-in__form"
      >
        <div
          class="sign-in__fields"
        >
          <div
            class="sign-in__field"
          >
            <input
              class="sign-in__input"
              id="user-email"
              name="email"
              placeholder="Email address"
              required=""
              type="email"
            />
            <label
              class="sign-in__label visually-hidden"
              for="user-email"
            >
              Email address
            </label>
          </div>
          <div
            class="sign-in__field"
          >
            <input
              class="sign-in__input"
              id="user-password"
              name="password"
              placeholder="Password"
              required=""
              type="password"
            />
            <label
              class="sign-in__label visually-hidden"
              for="user-password"
            >
              Password
            </label>
          </div>
        </div>
        <div
          class="sign-in__submit"
        >
          <button
            class="sign-in__btn"
            type="submit"
          >
            Sign in
          </button>
        </div>
      </form>
    </div>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link logo__link--light"
          href="main.html"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
